### PR TITLE
dynamic resource allocation: Improve code coverage of state checkpoint

### PR DESCRIPTION
**Pull request by tmilox**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

If applied this commit will increase the code coverage of `pkg/kubelet/cm/dra/state/state_checkpoint.go`

- before changes

Screenshot 2023-07-04 at 08 25 16

- after changes

Screenshot 2023-07-04 at 08 24 51

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/kubelet/cm/dra/state/state_checkpoint_test.go [hunk 1]
```diff
@@ -183,6 +183,11 @@ func TestCheckpointStateStore(t *testing.T) {
 
 	expectedCheckpoint := `{"version":"v1","entries":[{"DriverName":"test-driver.cdi.k8s.io","ClassName":"class-name","ClaimUID":"067798be-454e-4be4-9047-1aa06aea63f7","ClaimName":"example","Namespace":"default","PodUIDs":{"139cdb46-f989-4f17-9561-ca10cfb509a6":{}},"CDIDevices":{"test-driver.cdi.k8s.io":["example.com/example=cdi-example"]}}],"checksum":153446146}`
 
+	// should return an error, stateDir cannot be an empty string
+	if _, err := NewCheckpointState("", testingCheckpoint); err == nil {
+		t.Fatal("expected error but got nil")
+	}
+
 	// create temp dir
 	testingDir, err := os.MkdirTemp("", "dramanager_state_test")
```
[the remark this answers is not in the document]
Comment (TommyStarK) [reply pkg/kubelet/cm/dra/state/state_checkpoint_test.go:192]: as per the [comment](https://github.com/kubernetes/kubernetes/pull/119058/files#diff-e964495594ff51063ffb433ffea8b02b207a6b701c903f3f0ed62bca5cc3e607R186) passing an empty string to cause an error

[the remark this answers is not in the document]
Comment (charles-chenzz) [reply pkg/kubelet/cm/dra/state/state_checkpoint_test.go:192]: I didn't notice that, sorry

File: pkg/kubelet/cm/dra/state/state_checkpoint_test.go [hunk 2]
```diff
@@ -183,6 +183,11 @@ func TestCheckpointStateStore(t *testing.T) {
 
 	expectedCheckpoint := `{"version":"v1","entries":[{"DriverName":"test-driver.cdi.k8s.io","ClassName":"class-name","ClaimUID":"067798be-454e-4be4-9047-1aa06aea63f7","ClaimName":"example","Namespace":"default","PodUIDs":{"139cdb46-f989-4f17-9561-ca10cfb509a6":{}},"CDIDevices":{"test-driver.cdi.k8s.io":["example.com/example=cdi-example"]}}],"checksum":153446146}`
 
+	// should return an error, stateDir cannot be an empty string
```
Comment (klueska) [inline pkg/kubelet/cm/dra/state/state_checkpoint_test.go:186]:
```suggestion
	// Should return an error, stateDir cannot be an empty string
```

File: pkg/kubelet/cm/dra/state/state_checkpoint_test.go [hunk 3]
```diff
@@ -203,4 +208,13 @@ func TestCheckpointStateStore(t *testing.T) {
 	checkpointData, err := checkpoint.MarshalCheckpoint()
 	assert.NoError(t, err, "could not Marshal Checkpoint")
 	assert.Equal(t, expectedCheckpoint, string(checkpointData), "expected ClaimInfoState does not equal to restored one")
+
+	// checkpoint state with an empty checkpointName should return an error when trying to save state to checkpoint
+	cs, err = NewCheckpointState(testingDir, "")
+	if err != nil {
+		t.Fatal(err)
+	}
```
Comment (klueska) [inline pkg/kubelet/cm/dra/state/state_checkpoint_test.go:216]: I would argue that `NewCheckpointState()` itself should return an error if no checkpoint filename is passed. Feel free to make that change as part of this PR.

File: pkg/kubelet/cm/dra/state/state_checkpoint_test.go (hunk 2 above)
Comment (TommyStarK) [reply pkg/kubelet/cm/dra/state/state_checkpoint_test.go]: done in f924bf95df9

File: pkg/kubelet/cm/dra/state/state_checkpoint_test.go (hunk 3 above)
Comment (TommyStarK) [reply pkg/kubelet/cm/dra/state/state_checkpoint_test.go]: done in f924bf95df9

**Review by klueska — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**